### DISCUSSION TITLE
unable to delete user AuthApiError : "database error while deleting user"

### DISCUSSION THREAD

**Discussion by Qasim-De** [supabase/supabase#16232] [Questions]
I am try to implement a feature where user can have an option to delete there account. I follow the documentation and implemented using this code (https://supabase.com/docs/reference/javascript/auth-admin-deleteuser)

`const { data, error } = await supabase.auth.admin.deleteUser(
  '715ed5db-f090-4b8c-a067-640ecee36aa0'
)`

but in response supabase throw me error **AuthAPIError** "Failed to delete user database error". 

I try to delete user even from the dashboard > Authentication page but still having the same issue on both sides (Dashboard & SDK) 

Error screenshot attached 

<img width="1310" alt="Screenshot 2023-08-04 at 8 57 34 PM" src="https://github.com/supabase/supabase/assets/88767038/0dfbe7e3-1dae-450f-bd97-feddf61c8188">

Note : I have auth and profile tables in my database.

**Comment by GaryAustin1** [comment]
If you look in the Postgres logs is there any error right after the delete?  I would think the dashboard error you got would mention if you have other tables with data foreign key linked to the user, but maybe it changed.  That is the most common error for not being able to delete.

**Comment by Qasim-De** [comment]
i am getting these logs from logs > postgres. 

What is OLD in this query, here is my logs 

`{ "file": null, "host": "db-nrtalmafplukoisopcjr", "metadata": [], "parsed": [ { "application_name": null, "backend_type": "client backend", "command_tag": "DELETE", "connection_from": "127.0.0.1:49394", "context": "SQL statement \"DELETE FROM auth.users WHERE auth.users.id = OLD.user_id\"\nPL/pgSQL function public.deleteuser() line 3 at SQL statement", "database_name": "postgres", "detail": null, "error_severity": "ERROR", "hint": null, "internal_query": null, "internal_query_pos": null, "leader_pid": null, "location": null, "process_id": 189667, "query": "DELETE FROM \"users\" AS users WHERE users.id = $1", "query_id": 7231038676619066000, "query_pos": null, "session_id": "64ccf729.2e4e3", "session_line_num": 12, "session_start_time": "2023-08-04 13:03:37 UTC", "sql_state_code": "42703", "timestamp": "2023-08-07 07:42:44.801 UTC", "transaction_id": 3773, "user_name": "supabase_auth_admin", "virtual_transaction_id": "9/99114" } ], "parsed_from": null, "project": null, "source_type": null }`

**Comment by Qasim-De** [comment]
i am getting this logs under logs > Postgres, but unable to understand what
is OLD in this query ?

{ "file": null, "host": "db-nrtalmafplukoisopcjr", "metadata": [], "parsed":
 [ { "application_name": null, "backend_type": "client backend",
"command_tag": "DELETE", "connection_from": "127.0.0.1:49394", "context": "SQL
statement \"DELETE FROM auth.users WHERE auth.users.id =
OLD.user_id\"\nPL/pgSQL function public.deleteuser() line 3 at SQL
statement", "database_name": "postgres", "detail": null, "error_severity":
"ERROR", "hint": null, "internal_query": null, "internal_query_pos": null,
"leader_pid": null, "location": null, "process_id": 189667, "query": "DELETE
FROM \"users\" AS users WHERE users.id = $1", "query_id":
7231038676619066000, "query_pos": null, "session_id": "64ccf729.2e4e3",
"session_line_num": 12, "session_start_time": "2023-08-04 13:03:37 UTC",
"sql_state_code": "42703", "timestamp": "2023-08-07 07:42:44.801 UTC",
"transaction_id": 3773, "user_name": "supabase_auth_admin",
"virtual_transaction_id": "9/99114" } ], "parsed_from": null, "project":
null, "source_type": null }

**Reply by GaryAustin1** [reply]
Sorry just saw this...
It appears to me you have your own deleteuser function in the public schema.  Do you have a trigger function that calls this?

**Reply by athos-tolky** [reply]
I am having the same problem, did you know something we can do? 

version: 
<img width="1446" alt="image" src="https://github.com/user-attachments/assets/fa2b393a-6f8a-4a62-9aaa-b2285b687f80">

**Reply by GaryAustin1** [reply]
You should look at the Postgres and/or auth logs and see what is causing the error.  The most common case is you have tables belonging to the user with a foreign key and those have to be deleted first, or you have to have cascade delete for the foreign key constraint.

**Reply by athos-tolky** [reply]
> You should look at the Postgres and/or auth logs and see what is causing the error. The most common case is you have tables belonging to the user with a foreign key and those have to be deleted first, or you have to have cascade delete for the foreign key constraint.

this is happening with any user, and everything was working fine yesterday when i was deleting users. i didn’t change anything in the tables or make any similar modifications. i was just creating and deleting users in the authentication section on supabase to test the sso login feature on our website. when i run a direct query in the sql editor to delete a user, it works. the issue only occurs through the authentication panel.

error:
```
{"auth_event":{"action":"user_deleted","actor_id":"00000000-0000-0000-0000-000000000000","actor_username":"service_role","actor_via_sso":false,"log_type":"team","traits":{"user_email":"athosmartinezandrade2004@outlook.com","user_id":"9350e9f6-b122-4da5-b5c5-c97323579bd6","user_phone":""}},"component":"api","error":"ERROR: function unaccent(text) does not exist (SQLSTATE 42883)","level":"error","method":"DELETE","msg":"500: Database error deleting user","path":"/admin/users/9350e9f6-b122-4da5-b5c5-c97323579bd6","referer":"https://tolky.to","remote_addr":"186.194.175.173","request_id":"8dbc75ab91441a9e-GRU","time":"2024-11-01T14:08:57Z","user_id":"9350e9f6-b122-4da5-b5c5-c97323579bd6"}
```

**Reply by GaryAustin1** [reply]
That looks like the auth log.  Anything in the Postgres log?
Do you have a delete trigger function on anything?
The error is complaining about unaccent() being called.  That is an extension that is involved with text searches.  I don't think Supabase uses it by default, but not sure.

**Reply by athos-tolky** [reply]
> That looks like the auth log. Anything in the Postgres log? Do you have a delete trigger function on anything? The error is complaining about unaccent() being called. That is an extension that is involved with text searches. I don't think Supabase uses it by default, but not sure.

nothing at postgres log, only at auth log. we don’t have any delete trigger function on our database. i don’t know what we can do more, but it’s very strange because when i use sql queries in sql editor, the deletion works

<img width="1351" alt="image" src="https://github.com/user-attachments/assets/0708a005-5659-4615-ad12-f23d8dfff46c">

**Reply by GaryAustin1** [reply]
The auth log is pretty clear it is a database error and because of unaccent.  I have no idea if it is from your delete or something else though.  A database error will show up in the Postgres log at the same time.  You don't happen to have a filter on log type for the Postgres log do you?

Without more info not much users can help with.  You probably should try Supabase support.

**Reply by athos-tolky** [reply]
> The auth log is pretty clear it is a database error and because of unaccent. I have no idea if it is from your delete or something else though. A database error will show up in the Postgres log at the same time. You don't happen to have a filter on log type for the Postgres log do you?
> 
> Without more info not much users can help with. You probably should try Supabase support.

That’s my auth logs when I try to delete a user.

![image](https://github.com/user-attachments/assets/52be40ff-69cf-4e04-a116-d9e5491d0994)

**Reply by GaryAustin1** [reply]
Unless the Postgres log has more info on the error, that extension function is being called and is not found.  It is an optional extension so I assume you use it for something in the delete path.

**Reply by athos-tolky** [reply]
> Unless the Postgres log has more info on the error, that extension function is being called and is not found. It is an optional extension so I assume you use it for something in the delete path.

Thanks for the support! I solved the problem by turning off the unaccent extension and turning it on again in the auth schema, because it was turned on only in the public schema.

**Comment by abdrhxyii** [comment]
did you fix the issue ?

**Reply by yehdori** [reply]
You need to include the shouldSoftDelete value (which is boolean) in the parameters. 
I got the same error when I only included the userId, but once I added that value, it worked fine.

`const { data, error } = await supabase.auth.admin.deleteUser(userId, true/false)`
